lib/excon/socket.rb

```diff
@@ -304,17 +311,30 @@ def write_block(data)
     end
 
     def select_with_timeout(socket, type)
+      timeout_kind = type
+      timeout = @data[OPERATION_TO_TIMEOUT[type]]
+
+      if @data.include?(:deadline)
```

Comment:
I think it would be good to add a bit of a comment here to explain the logic/intention of this chunk of code, since it's important (and since I don't think I really fully understood it during the earlier review), and maybe the slight rename suggested below. I was thinking something like: `If a request timeout is set, use the smaller of the time remaining before the request timeout and the timeout for the type of select`.

Reply:
Thanks for pointing out that the code was not obvious. I added 2 comments: one above the deadline check to explain why we check for a deadline, the other above the check to see if we should use the request timeout (basically my own wording on your suggestion above).

I also renamed the `remaining` local variable to `request_timeout` to make it clearer what the value represents.

Reply:
Looks good, thanks!